src/js/components/ValidationProgress.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { PropTypes } from 'react'
+
+const ValidationProgress = (props) => {
+
+  const code = props.status.code
+
+  let uploadComplete = null
+  let parsingStarted = null
+  let parsingComplete = null
+  let validationStarted = null
+  let validationComplete = null
+
```

Comment:
Do you think we need a `if(code == 2) uploadingStarted = <p>Uploading ...</p>`?

And, I think the `parsingStarted` and `validationStarted` should be removed from the UI when the `Complete` state is reached. (Same for `uploadingStarted` if we add it.)